sdk/resourcemanager/Azure.ResourceManager.Core/tests/Scenario/HttpPipelineTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Identity;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests.Scenario
+{
+    public class HttpPipelineTests : ResourceManagerTestBase
+    {
+        private ArmClient _client;
+
+        public HttpPipelineTests(bool isAsync)
+            : base(isAsync) //, RecordedTestMode.Record)
+        {
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            _client = GetArmClient();
+        }
+
+        [Test]
+        public void ValidateHttpPipelines()
+        {
+            ResourceGroup resourceGroup = _client.DefaultSubscription
+                .GetResourceGroups().Construct("westus")
+                .CreateOrUpdateAsync("test-CacheHttpPipeline").ConfigureAwait(false).GetAwaiter().GetResult();
+            Assert.AreEqual(resourceGroup.Pipeline.GetHashCode(), _client.DefaultSubscription.Pipeline.GetHashCode());
```

Comment:
Can we add a paging method and do the same assert?

foreach(var rg in _client.DefaultSubscription.GetResourceGroups().List())
{
  Assert.AreEqual(rg.Pipeline.GetHashCode(), _client.DefaultSubscription.Pipeline.GetHashCode());
}